libensemble/manager.py

```diff
@@ -597,7 +652,7 @@ def _alloc_work(self, H: npt.NDArray, persis_info: dict) -> dict:
 
         alloc_f = self.alloc_specs["alloc_f"]
         output = alloc_f(
-            self.W,
+            self.W.iterable,
```

Comment:
I'm uneasy about this. Should we not use self.W in the same way in alloc funcs (inc. in alloc support). Esp. as there is potentially invisible errors if you end up with W[-1]

Reply:
Good point. I'll refactor more of alloc_support

Reply:
One complication I just realized is that that'll expose the _WorkerIndexer wrapper to the user. `W` as traditionally within the alloc_f would no longer be be a NumPy array unless we pass in the iterable as we do here.

Reply:
Ah, but if we don't explicitly use W outside of persistent_support functions, then we can wrap it with an indexer inside PersistentSupport

Reply:
I'm leaning towards we pass self.W, even if its breaking. I will review the alloc funcs.